### PR TITLE
fix(capecod): include on-level factor in exposure weighting (#1308, #1319)

### LINKED ISSUE #1308: [BUG] Potential Bug in Capecod
### Are you on the latest chainladder version?

- [x] Yes, this bug occurs on the latest version.

### Describe the bug in words

[This line](https://github.com/casact/chainladder-python/blob/91a942f1d5fad786accc654968c08b4f0fda0f80/chainladder/methods/capecod.py#L270) doesn't read right to me.
```
detrended_apriori_ = apriori_ / trend_array / X_olf_array * sw_olf_array
```

 `X_olf_array` and `sw_olf_array` should be identical, meaning multiplying by one then immediately dividing by the other nets out to be no actual onleveling being applied. this potential bug is currently unverified because `Capecode._onlevel` is uncovered for when olf_ exists. 

@priyam0k can you please refactor the olf example in Chapter 10 to route through `Capecode._onlevel` and validate this calculation?

### How can the bug be reproduced?

this should be reproduceable through chapter 10 of friedland

### What is the expected behavior?

N/A

### Would you be willing to contribute this ticket?

- [ ] Yes, absolutely!
- [ ] Yes, but I would like some help.
- [ ] No.

### LINKED ISSUE #1319: [REFACTOR] Friedland Chapter 10 CapeCod on-leveling and add tests
### Summary
In Friedland Chapter 10 (Exhibit II), the on-level adjustments and detrending are currently done manually in pandas instead of letting `CapeCod` handle them. 

Following up on #1308, we want to route this example cleanly through `CapeCod` and add proper test coverage.

### What needs to be done
- Fix the exposure weighting in `capecod.py` ([line 260](https://github.com/casact/chainladder-python/blob/91a942f1d5fad786accc654968c08b4f0fda0f80/chainladder/methods/capecod.py#L261-L263) ) so it includes the on-level factors.
- Update Chapter 10 Exhibit II to use `CapeCod` directly without the manual workarounds.
- Add a unit test to make sure CapeCod with on-level factors stays covered.

### PR DESCRIPTION
**Pull request by priyam0k**
## Summary of Changes  
- Included sample weight on-level factors in the Cape Cod weighted exposure calculation so premium rate changes are properly reflected in the a priori loss ratio.
- Updated Friedland Chapter 10 Exhibit II to use native CapeCod on-level functionality directly in the pipeline instead of manual pandas steps.
- Fixed notebook lint issues in Chapter 10 and removed its grandfathered ignore entry from pyproject.toml.
- Added a unit test validating Cape Cod with both rate level changes and tort reform against textbook values.

## Related GitHub Issue(s)  
- Fixes #1308
- Closes #1319
- Helps with #1216

## Additional Context for Reviewers  
All pre-commit ruff checks and tests pass cleanly.

## Checklist
- [x] I passed tests locally for both code (uv run pytest) and documentation changes (uv run --directory docs jb build . --builder=custom --custom-builder=doctest)

---

> [!NOTE]
> **Medium Risk**
> Changes core Cape Cod apriori math when exposure carries `olf_` (e.g. on-leveled premium), which can shift reserves and loss ratios for those workflows; scope is limited to one formula line plus regression tests.
> 
> **Overview**
> Fixes **Cape Cod** so exposure used to weight the all-years a priori loss ratio includes **sample-weight on-level factors** (`sw_olf_array`), aligning weighted exposure with how premium is already scaled in the loss-ratio numerator/denominator.
> 
> Adds **`test_capecod_onlevel_friedland_exhibit_ii`** (rate on-level + tort reform pipeline vs. Friedland Ch. 10 Exhibit II). Updates **`docs/friedland/chapter_10.ipynb`** Exhibit II to drive projections from `CapeCod` outputs (`detrended_apriori_`, `expectation_`, `ibnr_`, `ultimate_`) and on-level premium via `ParallelogramOLF.fit_transform`, with ruff cleanup and removal of the notebook’s **`pyproject.toml`** per-file ignore.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 98af00cdcb5d7fdec52ea2db6b11fb1cbe0fcc5f. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: chainladder/methods/tests/test_capecod.py [hunk 1]
```diff
@@ -133,3 +133,92 @@ def test_capecod_predict_one_extra_index_level(clrd):
...
+            "1/1/2005",
+            "1/1/2006",
+            "1/1/2007",
+            "1/1/2008",
+        ],
+        "rate_change": [0.02, 0.02, 0.02, 0.02, 0.05, 0.075, 0.15, 0.10, -0.20, -0.20],
+    })
+    tort_history = pd.DataFrame({
+        "date": ["1/1/2006", "1/1/2007"],
+        "rate_change": [-0.1067, -0.25],
```
Comment (henrydingliu) [inline chainladder/methods/tests/test_capecod.py]: we can use your cumulative enhancement to parallelogramolf here

File: chainladder/methods/tests/test_capecod.py [hunk 2]
```diff
@@ -133,3 +133,92 @@ def test_capecod_predict_one_extra_index_level(clrd):
...
+    })
+
+    onlevel = cl.ParallelogramOLF(
+        rate_history, change_col="rate_change", date_col="date", vertical_line=True
+    )
+    tort = cl.ParallelogramOLF(
+        tort_history, change_col="rate_change", date_col="date", vertical_line=True
+    )
+
+    floored_patterns = dict(
```
Comment (henrydingliu) [inline chainladder/methods/tests/test_capecod.py:188]: try `Triangle.maximum`

File: chainladder/methods/tests/test_capecod.py [hunk 3]
```diff
@@ -133,3 +133,92 @@ def test_capecod_predict_one_extra_index_level(clrd):
...
+        )
+    )
+
+    sample_weight = onlevel.fit_transform(xyz_premium)
+    cc = (
+        cl
+        .Pipeline([
+            ("tort", tort),
+            ("dev", cl.DevelopmentConstant(patterns=floored_patterns, style="cdf")),
+            ("capecod", cl.CapeCod(trend=0.0342)),
```
Comment (henrydingliu) [inline chainladder/methods/tests/test_capecod.py]: the .00005 rounding on trend actually makes a difference

**Comment by priyam0k** [comment]
This is ready for review.

File: chainladder/methods/tests/test_capecod.py (hunk 2 above)
Comment (henrydingliu) [reply chainladder/methods/tests/test_capecod.py:188]: did you miss this one?

Comment (priyam0k) [reply chainladder/methods/tests/test_capecod.py:188]: Updated [here](https://github.com/casact/chainladder-python/blob/a98795e4d8fdb264a7a717a7f9858865e6caacd0/chainladder/methods/tests/test_capecod.py#L187-L190) with floored_cdf = xyz_reported_dev.cdf_.maximum(1.0) .  missed resolving the thread

**Comment by henrydingliu** [comment]
please resolve conflict

**Review by henrydingliu — APPROVED** [review]
